### PR TITLE
CASMUSER-3008: Initial support for installing UAIs on UAN

### PR DESCRIPTION
**Pull request by alanm-hpe**
## Summary and Scope

CASMUSER-3008: Initial support for installing k3s, metallb, haproxy, sshd

Add roles necessary to support UAIs on UAN. This includes:
uan_k3s_stage   - Pull k3s assets to the image
uan_k3s_install - Install k3s master on a node
uan_helm        - Initialize environment for helm
uan_metallb     - Install MetalLB chart
uan_haproxy     - Install HAProxy charts
uan_sshd        - Install and configure SSHD instances

To use the new roles, a playbook "k3s.yml" has been added. By default, MetalLB will be installed without an IPAddressPool unless configuration action is taken.

Additionally, an example HAProxy configuration is included to show a default SSH load balancer that can be further extended to support UAIs using podman.

## Issues and Related PRs

* Resolves [CASMUSER-3008](https://jira-pro.its.hpecorp.net:8443/browse/CASMUSER-3008)

## Testing

  * `surtur`
  * Local development environment

### Test description:

Ansible was tested manually on the UAN in addition to running CFS Image Customization and Image Personalization

## Risks and Mitigations

This feature is considered a Technical Preview, so the configs are likely to need changes or be expanded on.
As the playbook is not run by default, the risk to impacting nominal UAN installations is zero

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by alanm-hpe** [comment]
Testing results:
```
uan01:~ # export KUBECONFIG=~/.kube/k3s.yml

uan01:~ # kubectl get nodes -A
NAME    STATUS   ROLES                  AGE   VERSION
uan01   Ready    control-plane,master   55m   v1.26.0+k3s1

uan01:~ # helm ls -A
NAME       	NAMESPACE     	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
haproxy-gpu	haproxy-gpu   	1       	2023-01-19 09:33:21.596715903 -0600 CST	deployed	haproxy-1.17.3	2.6.6
haproxy-uai	haproxy-uai   	1       	2023-01-19 09:21:42.321192353 -0600 CST	deployed	haproxy-1.17.3	2.6.6
metallb    	metallb-system	1       	2023-01-19 09:21:16.180300139 -0600 CST	deployed	metallb-0.13.7	v0.13.7

uan01:~ # kubectl get pods,services -n haproxy-gpu
NAME                    READY   STATUS    RESTARTS   AGE
pod/haproxy-gpu-2ls4l   1/1     Running   0          43m

NAME                  TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)        AGE
service/haproxy-gpu   LoadBalancer   10.43.225.10   10.103.11.177   22:30987/TCP   43m
```

**Comment by alanm-hpe** [comment]
Using some extra configuration for MetalLB, HAProxy, and SSHD, I was able to test an SSH load balancer through HAProxy and into a new SSHD instance that can eventually be configured for podman:
```
arbus ~/github/uan (k3s) $ ssh 10.103.11.177
(alanm@10.103.11.177) Password:
Your password will expire in 279 day(s).
Your password will expire in 279 day(s).
Last login: Thu Jan 19 09:21:51 2023 from 10.42.0.7
alanm@uan01:~>
```

File: ansible/roles/uan_haproxy/templates/haproxy-values.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  listen ssh
+    bind *:22
+    balance leastconn
+    mode tcp
+
+    option tcp-check
+    tcp-check expect rstring SSH-2.0-OpenSSH.*
+
+    server uan01 10.103.11.142:9001 check inter 10s fall 2 rise 1
```
Comment (keopp-hpe) [inline ansible/roles/uan_haproxy/templates/haproxy-values.yml.j2:33]: Just wondering if the IP address here is always going to be this value or whether it should be a parameter?

Comment (alanm-hpe) [reply ansible/roles/uan_haproxy/templates/haproxy-values.yml.j2:33]: That shouldn't be there. Good catch. This is the default config block if nothing is defined. I do need a way to correctly populate this with values for a real uan though based on the ansible facts.

**Review by keopp-hpe — APPROVED** [review]

**Review by keopp-hpe — APPROVED** [review]

**Pull request merged**